### PR TITLE
refactor: remove heapless dependency

### PR DESCRIPTION
**Pull request by GabrielDertoni**
This PR aims to remove the [`heapless`](https://crates.io/crates/heapless) dependency. Currently the implementation relies on `heapless::Vec` but doesn't have to. Removing this dependency reduces the ammount of code generated and the compile time. It also uses a `const DEFAULT` for default values of `*_DATA` types which can be used in `const fn`s.

Look at #156 for more context.

### CODE REVIEW

File: build/parser.rs [hunk 1]
```diff
@@ -179,7 +179,6 @@ impl MavProfile {
             #[allow(unused_imports)]
             use bitflags::bitflags;
             #[allow(unused_imports)]
```
Comment (danieleades) [inline build/parser.rs]: do you need to remove this flag?

File: build/parser.rs [hunk 2]
```diff
@@ -582,12 +597,34 @@ impl MavMessage {
...
+            }
+        }
+    }
+
+    fn emit_const_default(&self) -> TokenStream {
+        let initializers = self
+            .fields
+            .iter()
+            .map(|field| field.emit_default_initializer())
+            .collect::<TokenStream>();
```
Comment (danieleades) [inline build/parser.rs]: does this work without the `collect`?

File: build/parser.rs [hunk 3]
```diff
@@ -806,23 +861,11 @@ impl MavType {
...
-                    // handle as a slice
-                    let r = t.rust_reader(&quote!(let val), buf);
-                    quote! {
-                        for idx in 0..#size {
-                            #r
-                            #val[idx] = val;
-                        }
+                let r = t.rust_reader(&quote!(let val), buf);
+                quote! {
+                    for idx in 0..#size {
```
Comment (danieleades) [inline build/parser.rs]:
```suggestion
                    for v in &mut val {
```

(or something similar)

File: build/parser.rs [hunk 4]
```diff
@@ -733,6 +771,21 @@ impl MavField {
...
         }
     }
+
+    fn emit_default_initializer(&self) -> TokenStream {
+        let field = self.emit_name();
+        // FIXME: Is this actually expected behaviour??
+        if matches!(self.mavtype, MavType::Array(_, _)) {
+            let default_value = self.mavtype.emit_default_value();
+            quote!(#field: #default_value,)
+        } else if let Some(ref enumname) = self.enumtype {
```
Comment (danieleades) [inline build/parser.rs]:
```suggestion
        } else if let Some(enumname) = &self.enumtype {
```

can match ergonomics help here?

**Comment by patrickelectric** [comment]
@GabrielDertoni check ci

**Comment by qwerty19106** [comment]
You can replace [serde_arrays](https://github.com/Kromey/serde_arrays) dependency on [serde-big-array](https://github.com/est31/serde-big-array) to fix `the trait 'Default' is not implemented for '[u8; 200]'` problem.

**Review by patrickelectric — APPROVED** [review]

**Comment by patrickelectric** [comment]
It appears fine by be, is there any further comments around this PR ?

**Comment by qwerty19106** [comment]
It would be nice to add a link to the issue into the `RustDefault` description.
The best I found is [issue](https://users.rust-lang.org/t/issue-for-derives-for-arrays-greater-than-size-32/59055/3).

> Default is special, because we already have impl<T> Default for [T; 0] without the T: Default constraint. Rewrite it in const generics requires the const generics specialization, which is far from the stable yet.

**Comment by GabrielDertoni** [comment]
ping @patrickelectric

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**